sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationkeyvaultoptions.py

```diff
@@ -3,27 +3,51 @@
-
+from typing import overload, List, Optional, Callable
+from azure.keyvault.secrets import SecretClient
 
 class AzureAppConfigurationKeyVaultOptions:
-    """
-    Options for connecting to Key Vault.
 
-    :param credential: A credential for authenticating with the key vault. This is optional if secret_clients is
-     provided.
-    :type credential: ~azure.core.credentials.TokenCredential
-    :param secret_clients: A list of SecretClient from azure-keyvault-secrets. This is optional if credential is
-     provided.
-    :type secret_clients: list[~azure.keyvault.secrets.SecretClient]
-    :param secret_resolver: A function that takes a URI and returns a value.
-    :type secret_resolver: callable
-    """
+    @overload
+    def __init__(self,  credential, *, secret_clients: Optional[List[SecretClient]] = None):
+        """
+        Options for connecting to Key Vault.
+
+        :param credential: A credential for authenticating with the key vault. This is optional if secret_clients is
+        provided.
+        :type credential: ~azure.core.credentials.TokenCredential
+        :param secret_clients: A list of SecretClient from azure-keyvault-secrets. This is optional if credential is
+        provided.
+        :type secret_clients: list[~azure.keyvault.secrets.SecretClient]
+        """
+        ...
+
+    @overload
+    def __init__(self, secret_clients: List[SecretClient], * , secret_resolver: Optional[Callable[[str], str]] = None):
+        """
+        Options for connecting to Key Vault.
+
+        :param secret_clients: A list of SecretClient from azure-keyvault-secrets. This is optional if credential is
+        provided.
+        :type secret_clients: list[~azure.keyvault.secrets.SecretClient]
+        :param secret_resolver: A function that takes a URI and returns a value.
+        :type secret_resolver: Callable[[str], str]
+        """
+        ...
 
-    def __init__(self, credential=None, secret_clients=None, secret_resolver=None):
-        # type: (TokenCredential, List[SecretClient], Callable) -> None
-        self.credential = credential
-        self.secret_clients = secret_clients
-        self.secret_resolver = secret_resolver
+    def __init__(self, **kwargs):
+        """
+        Options for connecting to Key Vault.
 
-        if self.secret_clients is None:
-            self.secret_clients = {}
+        :param credential: A credential for authenticating with the key vault. This is optional if secret_clients is
+        provided.
+        :type credential: ~azure.core.credentials.TokenCredential
+        :param secret_clients: A list of SecretClient from azure-keyvault-secrets. This is optional if credential is
+        provided.
+        :type secret_clients: list[~azure.keyvault.secrets.SecretClient]
+        :param secret_resolver: A function that takes a URI and returns a value.
+        :type secret_resolver: Callable[[str], str]
+        """
+        self.credential = kwargs.get("credential", None)
```

Comment:
Should we validate to make sure credential and secret_resolver both cannot be provided together?

Reply:
Added.